easy-ui-react/src/PlanCard/PlanCard.tsx

```diff
@@ -0,0 +1,217 @@
+import React, { ReactNode, useRef } from "react";
+import { AriaLinkOptions, mergeProps, useHover, useLink } from "react-aria";
+import { Card, CardProps } from "../Card";
+import { Text, TextProps } from "../Text";
+import { VerticalStack } from "../VerticalStack";
+import { UnstyledButton, UnstyledButtonProps } from "../UnstyledButton";
+
+import styles from "./PlanCard.module.scss";
+
+type ChildrenFn = ({ isHovered }: { isHovered: boolean }) => ReactNode;
+
+export type PlanCardLinkProps = Omit<AriaLinkOptions, "children"> & {
+  /**
+   * The children of the `PlanCard.Link` element.
+   */
+  children: ChildrenFn | ReactNode;
+};
+
+export type PlanCardButtonProps = Omit<UnstyledButtonProps, "children"> & {
+  /**
+   * The children of the `PlanCard.Button` element.
+   */
+  children: ChildrenFn | ReactNode;
+};
+
+export type PlanCardProps = CardProps & {
+  /**
+   * The children of the `PlanCard` element.
+   */
+  children: ReactNode;
```

Comment:
Thought:
One thing (that probably shouldn't be addressed here) that we should probably think about is allowing multiple children. We could do like a `type ChildOrChildren = ReactNode | ReactNode[];` that can be used in all these places that don't really care.

Just a thought lol

Reply:
oh does `ReactNode` not cover `ReactNode[]`?

https://reactnative.dev/docs/react-node

Reply:
Nevermind, I guess that's not a thing 😅

I could've sworn in the past I have had to do stuff like that. Please disregard.